### ISSUE TITLE
Add an "eject" feature

### ISSUE THREAD

**Issue by fabiospampinato** [electron-userland/electron-webpack#318]
[create-react-app](https://github.com/facebook/create-react-app) has an "eject" command, where one basically calls "create-react-app eject" (or something similar, I haven't actually used this) and all the dynamic configuration that create-react-app creates dynamically for you are written to disk, so that eventually when your project grows and you want more control over your build chain you can have it pretty easily.

I think a similar command for `electron-webpack` where it outputs to disk the generate webpack configurations for the main process and the renderer process both for development and production would be great.

**Comment by loopmode** [comment]
That's a good idea! 

Btw, I started work on [create-electron-app](https://github.com/loopmode/create-electron-app) and I'd love the feature in there as well.
It would be a question in interactive mode and a CLI option in silent mode, e.g `yarn create @loopmode/electron-app my-app --eject --yes`.
I'll see what needs to be done in electron-webpack itself and what can be done externally.

**Comment by loopmode** [comment]
@fabiospampinato I've had no progress whatsoever, but until we figure something out, you do know how to do this easily manually, right?
Since we support the function as config, that is the final step - you can just dump the config object into a file manually, it contains any magic done by webpack. Let me know if I should provide more detail!

**Comment by fabiospampinato** [comment]
Yeah yeah I've actually reconfigured my the build chain already, thanks for asking! Dumping the default config is what I started up with, but that gives you 4 configuration files which are largely overlapping, and it's not immediately obvious what options have been passed to each plugin, as you see defaults too, but overall it's just a matter of browsing electron-webpack's source a bit.

---

While we're at it let me share my current configuration as an attachment to this post, perhaps it could be useful to other people who are trying to do the same thing, it requires some adaptations but it should provide a good starting point.

It may also be a decent starting point for rewriting `electron-webpack`, I find the current codebase to be not very maintainable.

My configuration kinda works the same way, but:

- Now my CSS files are hot-reloaded, I've no idea why this wasn't working before.
- The output to stdout is so much cleaner.
- There are no leftover processes when quitting the app.
- The app can be restarted by typing "rs" in the terminal, which is super useful since sometimes for some reason Electron just crashes at startup.
- There's a DLL bundle, which speeds things up massively during development (~3s vs ~600ms rebuilds).
- There are no extra external dependencies set, which shrinks the size of `app.asar`.
- Windows' bundles are generated also for the 32-bit architecture.
- MacOS' bundles are notarized.

[notable-webpack.zip](https://github.com/electron-userland/electron-webpack/files/3575288/notable-webpack.zip)
